jbmc/src/java_bytecode/README.md

```diff
@@ -39,9 +39,160 @@ To be documented.
 
 To be documented.
 
-\section java-bytecode-remove-exceptions remove_exceptions
+\section java-bytecode-remove-exceptions Remove exceptions
 
-To be documented.
+When `remove_exceptions` is called on the goto model, the goto model contains
+complex instructions such as `CATCH-POP`, `CATCH-PUSH` and `THROW`. In order to
+analyze the goto model, the instructions must be simplified to use more basic
+instructions - this is called "lowering". This class lowers the `CATCH` and
+`THROW` instructions.
+
+Exceptions that have been thrown, but not yet caught are stored the global
+`@inflight_exception`. When they have been caught, the global
+`@inflight_exception` is set back to null and ready for the next exception.
+
+\subsection Throw
+
+Consider a simple method `testException(I)V`:
+
+```java
+public void testException(int i) throws Exception {
+    if (i < 0) {
+        throw new Exception();
+    }
+    field = i;
+}
+```
+
+The goto for `testException(I)V` before `remove_exceptions` (removing comments
+and replacing irrelevant parts with `...`) is:
+
+```
+com.diffblue.regression.TestExceptions.testException(int) /* java::com.diffblue.regression.TestExceptions.testException:(I)V */
```

Comment:
there is no package in the example above. I suggest to remove it here.

Reply:
The package is used below, so I will leave it.

Reply:
The maybe complete the example by wrapping it into a `class TestExceptions` and adding a `package` statement in the beginning.

Reply:
I've removed the package and included a copy-pasteable test.